Geometry/MTDNumberingBuilder/src/MTDTopology.cc

```diff
@@ -1,3 +1,135 @@
 #include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
-MTDTopology::MTDTopology(const int &topologyMode) : mtdTopologyMode_(topologyMode) {}
+MTDTopology::MTDTopology(const int& topologyMode, const ETLValues& etl)
+    : mtdTopologyMode_(topologyMode), etlVals_(etl) {}
+
+bool MTDTopology::orderETLSector(const GeomDet*& gd1, const GeomDet*& gd2) {
+  ETLDetId det1(gd1->geographicalId().rawId());
+  ETLDetId det2(gd2->geographicalId().rawId());
+
+  if (det1.mtdRR() != det2.mtdRR()) {
+    return det1.mtdRR() < det2.mtdRR();
+  } else if (det1.modType() != det2.modType()) {
+    return det1.modType() < det2.modType();
+  } else {
+    return det1.module() < det2.module();
+  }
+}
+
+size_t MTDTopology::hshiftETL(const uint32_t detid, const int horizontalShift) const {
+  static constexpr size_t failIndex(
+      std::numeric_limits<unsigned int>::max());  // return out-of-range value for any failure
+
+  ETLDetId start_mod(detid);
+
+  if (horizontalShift == 0) {
+    edm::LogWarning("MTDTopology") << "asking of a null horizotalShift in ETL";
+    return failIndex;
+  }
+  int hsh = horizontalShift > 0 ? 1 : -1;
+
+  int module = start_mod.module();
+  uint32_t modtyp = start_mod.modType();
+  uint32_t discside = start_mod.discSide();
+
+  // ilayout number coincides at present with disc face, use this
+
+  size_t iHome = (modtyp == etlVals_[discside].idDetType1_) ? 0 : 1;
+  size_t iLeft = (etlVals_[discside].idDetType1_ == 1) ? 0 : 1;
+
+  // for left type modules the position according to the default order is module - 1, for the rigth type modules the total number of left modules must be added
+
+  size_t nmodOffset = (modtyp == 1) ? 0 : etlVals_[discside].start_copy_[iLeft].back() - 1;
+
+  for (size_t iloop = 0; iloop < etlVals_[discside].start_copy_[iHome].size() - 1; iloop++) {
+    if (module >= etlVals_[discside].start_copy_[iHome][iloop] &&
+        module < etlVals_[discside].start_copy_[iHome][iloop + 1]) {
+      if (module + hsh >= etlVals_[discside].start_copy_[iHome][iloop] &&
+          module + hsh < etlVals_[discside].start_copy_[iHome][iloop + 1]) {
+        return module + hsh - 1 + nmodOffset;
+      }
+      break;
+    }
+  }
+
+  return failIndex;
+}
+
+size_t MTDTopology::vshiftETL(const uint32_t detid, const int verticalShift, size_t& closest) const {
+  size_t failIndex(std::numeric_limits<unsigned int>::max());  // return out-of-range value for any failure
```

Comment:
This `failIndex` should also be `static constexpr`. Or maybe make it a data member so there is just one.

Reply:
I missed it, a class member looks a simpler solution, given its use in multiple methods